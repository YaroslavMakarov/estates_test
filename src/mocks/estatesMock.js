export const estatesMock = {
    estates: {
        estates: [
            {
                'id': 1,
                'title': 'Shevchenko estate',
                'price': 30000,
                'address': {
                    'street': 'Melnychna',
                    'number': 12
                },
                'preview_img': 'https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg',
                'seller': 'Chack Norris',
                'description': 'The best from Chuck',
                'detailed_photos': [
                    'https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg',
                    'https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg',
                    'https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg'
                ]
            },
            {
                "id": 2,
                "title": "Yanukovich estate",
                "price": 3000000,
                "address": {
                    "street": "Prisident",
                    "number": 4
                },
                "preview_img": "https://biz.liga.net/images/general/2018/11/15/thumbnail-tw-20181115111659-8474.jpg?v=1542279867",
                "seller": "Chack Norris",
                "description": "The best from Chuck",
                "detailed_photos": [
                    "https://www.wtoc.com/resizer/ckPwhlUdYaFcLiBWMX0J033a0Ow=/1200x0/arc-anglerfish-arc2-prod-raycom.s3.amazonaws.com/public/JGJB35IW4JAMROXX3D3MMEF2SU.jpg",
                    "https://img3.goodfon.ru/wallpaper/nbig/c/f7/chuck-norris-chak-norris-legenda-aktior-vzgliad.jpg",
                    "https://cdn.flickeringmyth.com/wp-content/uploads/2020/03/chuck-norris.jpg"
                    ]
             },
        ],
    },
};