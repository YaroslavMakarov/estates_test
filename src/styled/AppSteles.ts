import styled from 'styled-components';

export const AppWrapper = styled.div`
    padding-top: 50px; 
`;