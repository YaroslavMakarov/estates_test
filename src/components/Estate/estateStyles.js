import styled from "styled-components";

export const EstateContainer = styled.div`
    box-sizing: border-box;
    padding: 15px;
`;
export const Img = styled.img.attrs(props => ({
    src: props.src
}))`
    display: block;
    width: 100%;
    height: 300px;
    object-fit: cover;
    margin-bottom: 10px;
`;
export const EstateDescription = styled.div`
    font-size: 16px;
    margin-bottom: 5px;
`;