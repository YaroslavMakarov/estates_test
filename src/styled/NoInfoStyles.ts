import styled from 'styled-components';

export const NoInfoContent = styled.div`
    font-size: 24px;
    font-wight: bold;
`;