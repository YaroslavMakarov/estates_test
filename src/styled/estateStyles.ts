import styled from 'styled-components';

export const EstateContainer = styled.div`
    box-sizing: border-box;
    padding: 15px;
`;
export const Img = styled.img.attrs(props => ({
    src: props.src
}))`
    display: block;
    width: 100%;
    height: ${props => props.id ? "400px" : "300px"};;
    object-fit: cover;
    margin-bottom: 10px;
`;
export const EstateDescription = styled.div`
    font-size: ${props => props.id ? "22px" : "16px"};
    margin-bottom: ${props => props.id ? "10px" : "5px"};
`;