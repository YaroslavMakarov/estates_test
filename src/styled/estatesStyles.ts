import styled from "styled-components";

export const EstatesWrapper = styled.div`
    display: grid;
    grid-template-columns: repeat(4, 250px);
    grid-gap: 15px;
    width: fit-content;
`; 