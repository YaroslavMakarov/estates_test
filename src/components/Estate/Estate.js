import React from "react";
import { EstateContainer, Img, EstateDescription } from "./estateStyles"

const Estate = ({ estate }) => {
    return (
        <EstateContainer>
            <Img src={estate.preview_img} />
            <EstateDescription>
                {`Name: ${estate.title}`}
            </EstateDescription>
            <EstateDescription>
                {`Price: ${estate.price}$`}
            </EstateDescription>
            <EstateDescription>
                {`Address: ${estate.address.street}, ${estate.address.number}`}
            </EstateDescription>
        </EstateContainer>
    )
};

export default Estate;
