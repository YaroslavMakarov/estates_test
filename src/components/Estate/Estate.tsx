import React from 'react';
import { Link } from 'react-router-dom';

import { EstateContainer, Img, EstateDescription } from '../../styled/estateStyles';

type Props = {
    estate: Estate;
};

const Estate: React.FC<Props> = ({ estate }) => {

    return (
        <EstateContainer
          as={Link}
          to={`/info/${estate.id}`}
        >
            <Img src={estate.preview_img} />
            <EstateDescription>
                {`Name: ${estate.title}`}
            </EstateDescription>
            <EstateDescription>
                {`Price: ${estate.price}$`}
            </EstateDescription>
            <EstateDescription>
                {`Address: ${estate.address.street}, ${estate.address.number}`}
            </EstateDescription>
        </EstateContainer>
    )
};

export default Estate;
