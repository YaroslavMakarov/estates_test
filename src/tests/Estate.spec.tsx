import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { estatesMock } from '../mocks/estatesMock';
import Estate from '../components/Estate';
import { EstateContainer, Img, EstateDescription } from '../styled/estateStyles';

describe('Estate component', () => {
    let component: ShallowWrapper;
    const estate = estatesMock.estates.estates[0];
    beforeEach(() => {
        component = shallow(<Estate estate={estate} />);
    });

    it('get snapShot Estate', () => {
        expect(component.debug()).toMatchSnapshot();
    })

    it('get Link attr', () => {
        const estateLink = component.find(EstateContainer);
        expect(estateLink.prop('to')).toBe(`/info/${estate.id}`);
    })

    it('get Img attr', () => {
        const estateLink = component.find(Img);
        expect(estateLink.prop('src')).toBe(`${estate.preview_img}`);
    })

    it('repeating test for EstateDescription', () => {
        component.find(EstateDescription).forEach((node, index) => {
            const nodeNames = ['Name', 'Price', 'Address'];
            switch(index) {
                case 0: expect(node.text()).toBe(`${nodeNames[index]}: ${estate.title}`);
                break;
                case 1: expect(node.text()).toBe(`${nodeNames[index]}: ${estate.price}$`);
                break;
                case 2: expect(node.text()).toBe(`${nodeNames[index]}: ${estate.address.street}, ${estate.address.number}`);
                break;
            }                
        });
    });
})