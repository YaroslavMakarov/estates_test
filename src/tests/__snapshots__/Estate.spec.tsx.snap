// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Estate component get snapShot Estate 1`] = `
"<styled.div as={{...}} to=\\"/info/1\\">
  <styled.img src=\\"https://cp12.nevsepic.com.ua/62/1353962605-0435309-www.nevsepic.com.ua.jpg\\" />
  <styled.div>
    Name: Shevchenko estate
  </styled.div>
  <styled.div>
    Price: 30000$
  </styled.div>
  <styled.div>
    Address: Melnychna, 12
  </styled.div>
</styled.div>"
`;
